tests/tests.go

```diff
@@ -105,15 +106,36 @@ func SetupLocalRegistry(o *option.Option) {
 		_, name, _ := strings.Cut(ref, "/")
 		// allow up to a minute for remote pulls to account for external network
 		// latency/throughput issues or throttling (default is 10 seconds)
-		// retry pull for 3 times
+		// retry pull for 3 times.
 		var session *gexec.Session
+		exitCode := -1
 		for i := 0; i < retryPull; i++ {
-			session = command.New(o, "pull", ref).WithTimeoutInSeconds(30).WithoutCheckingExitCode().Run()
-			if session.ExitCode() == 0 {
+			session = command.New(o, "pull", ref).WithoutWait().WithoutCheckingExitCode().Run()
```

Comment:
You actually just need `WithoutWait()` here. Adding `WithoutCheckingExitCode` sets
```go
c.shouldWait = true
```
and `WithoutWait` exits earlier than `WithoutCheckingExitCode` anyway ([see here](https://github.com/runfinch/common-tests/blob/37fc4ecb301b5c02c62975b9506671f7ad05aa26/command/command.go#L112-L120)):
```go
	if !c.shouldWait {
		return session
	}
	session.Wait(c.timeout)

	if !c.shouldCheckExitCode {
		return session
	}
```

Reply:
i did think about it, just its more explicit while we read the code.

Reply:
It's more explicit but it's broken in this case because `WithoutCheckingExitCode` sets `shouldWait` to `true` which makes `Run` call `session.Wait`, which makes this code blocking.

It might not matter for your initial approach since you're polling, but if you do something more similar to my suggestion you don't even need to poll since `gexec.Start` (called in `Command.Run`) doesn't block.

Reply:
yes correct, updated it.